File: Data Structures & Algorithms/house-robber-ii/submission-1.java
class Solution {
    public int rob(int[] nums) 
    {
        int n = nums.length;
        
        if(n == 1) return nums[0];
        int dp[] = new int[n];
        dp[0] = nums[0];
        dp[1] = Math.max(nums[0],nums[1]);
        for(int i = 2;i<nums.length-1;i++){
            dp[i] = Math.max(dp[i-2]+nums[i],dp[i-1]);
        }
        int dp1[] = new int[n];
        dp1[0] = 0;
        dp1[1] = nums[1];
        for(int i = 2;i<nums.length;i++){
            dp1[i] = Math.max(dp1[i-2]+nums[i],dp1[i-1]);
        }
        return Math.max(dp[n-2],dp1[n-1]);
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-1.py
import math
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        #range of k -> 1 to max in the pile
        ans = float('inf')
        low = 1
        high = max(piles)
        while low <= high:
            mid = int((low+high)/2)
            hours = self.calculate_hours(piles, mid)
            print(mid)
            if hours <= h:
                ans = min(ans, mid)
                high = mid-1
            else:
                low = mid + 1
        
        return ans

        
    def calculate_hours(self, piles, k):
        hours = 0
        for banana in piles:
            hours += math.ceil(banana/k)
        # print(hours)
        return hours
        